[ICE]: discriminant_for_variant() is None

<!--
[31mICE[0m: Rustc ./a.rs '' 'thread 'rustc' ($TID) panicked at compiler/rustc_ty_utils/src/layout/invariant.rs:284:76: 'called `Option::unwrap()` on a `None` value'', 'thread 'rustc' ($TID) panicked at compiler/rustc_ty_utils/src/layout/invariant.rs:284:76: 'called `Option::unwrap()` on a `None` value''
File: /tmp/im2/a.rs
-->

auto-reduced (treereduce-rust):
````rust
const None: Option<unsafe<> Option<Box<dyn Send>>> = None;
fn main() {
    match None {
        _ => {}
    }
}

````

original:
````rust
#![feature(unsafe_binders)]
const None: Option<unsafe<> Option<Box<dyn Send>>> = None;
fn main() {
    match None {
        _ => {}
    }
}

````

Version information
````
rustc 1.96.0-nightly (3a933e5bf 2026-03-26)
binary: rustc
commit-hash: 3a933e5bf5b695faf9e09d47a263a20c6b85c26a
commit-date: 2026-03-26
host: x86_64-unknown-linux-gnu
release: 1.96.0-nightly
LLVM version: 22.1.2

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_ty_utils/src/layout/invariant.rs#L278-L290

Command:
`/home/matthias/.rustup/toolchains/alt-master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0658]: unsafe binder types are experimental
 --> /tmp/icemaker_global_tempdir.1IPDmI6BJABv/rustc_testrunner_tmpdir_reporting.9MTStWtlWgHl/mvce.rs:1:20
  |
1 | const None: Option<unsafe<> Option<Box<dyn Send>>> = None;
  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #130516 <https://github.com/rust-lang/rust/issues/130516> for more information
  = help: add `#![feature(unsafe_binders)]` to the crate attributes to enable
  = note: this compiler was built on 2026-03-26; consider upgrading it if it is out of date

error[E0277]: the trait bound `Box<(dyn Send + 'static)>: Copy` is not satisfied
 --> /tmp/icemaker_global_tempdir.1IPDmI6BJABv/rustc_testrunner_tmpdir_reporting.9MTStWtlWgHl/mvce.rs:1:13
  |
1 | const None: Option<unsafe<> Option<Box<dyn Send>>> = None;
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `Box<(dyn Send + 'static)>`
  |
  = note: required for `Option<Box<(dyn Send + 'static)>>` to implement `Copy`

error[E0277]: the trait bound `Box<(dyn Send + 'static)>: Copy` is not satisfied
 --> /tmp/icemaker_global_tempdir.1IPDmI6BJABv/rustc_testrunner_tmpdir_reporting.9MTStWtlWgHl/mvce.rs:1:54
  |
1 | const None: Option<unsafe<> Option<Box<dyn Send>>> = None;
  |                                                      ^^^^ the trait `Copy` is not implemented for `Box<(dyn Send + 'static)>`
  |
  = note: required for `Option<Box<(dyn Send + 'static)>>` to implement `Copy`


thread 'rustc' (3844251) panicked at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_ty_utils/src/layout/invariant.rs:284:76:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0:     0x7fd2509437b0 - std[4f619bdb2e663cc]::backtrace_rs::backtrace::libunwind::trace
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/../../backtrace/src/backtrace/libunwind.rs:117:9
   1:     0x7fd2509437b0 - std[4f619bdb2e663cc]::backtrace_rs::backtrace::trace_unsynchronized::<std[4f619bdb2e663cc]::sys::backtrace::_print_fmt::{closure#1}>
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/../../backtrace/src/backtrace/mod.rs:66:14
   2:     0x7fd2509437b0 - std[4f619bdb2e663cc]::sys::backtrace::_print_fmt
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/sys/backtrace.rs:74:9
   3:     0x7fd2509437b0 - <<std[4f619bdb2e663cc]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[e58e1890be41adad]::fmt::Display>::fmt
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/sys/backtrace.rs:44:26
   4:     0x7fd24bee6cbe - <core[e58e1890be41adad]::fmt::rt::Argument>::fmt
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/fmt/rt.rs:152:76
   5:     0x7fd24bee6cbe - core[e58e1890be41adad]::fmt::write
   6:     0x7fd25095b8dc - std[4f619bdb2e663cc]::io::default_write_fmt::<std[4f619bdb2e663cc]::sys::stdio::unix::Stderr>
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/io/mod.rs:639:11
   7:     0x7fd25095b8dc - <std[4f619bdb2e663cc]::sys::stdio::unix::Stderr as std[4f619bdb2e663cc]::io::Write>::write_fmt
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/io/mod.rs:1994:13
   8:     0x7fd250908286 - <std[4f619bdb2e663cc]::sys::backtrace::BacktraceLock>::print
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/sys/backtrace.rs:47:9
   9:     0x7fd250908286 - std[4f619bdb2e663cc]::panicking::default_hook::{closure#0}
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:292:27
  10:     0x7fd2509324a1 - std[4f619bdb2e663cc]::panicking::default_hook
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:319:9
  11:     0x7fd24d3d4c27 - <alloc[93d8057a61472300]::boxed::Box<dyn for<'a, 'b> core[e58e1890be41adad]::ops::function::Fn<(&'a std[4f619bdb2e663cc]::panic::PanicHookInfo<'b>,), Output = ()> + core[e58e1890be41adad]::marker::Send + core[e58e1890be41adad]::marker::Sync> as core[e58e1890be41adad]::ops::function::Fn<(&std[4f619bdb2e663cc]::panic::PanicHookInfo,)>>::call
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/alloc/src/boxed.rs:2254:9
  12:     0x7fd24d3d4c27 - rustc_driver_impl[c9be22246af3610]::install_ice_hook::{closure#1}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_driver_impl/src/lib.rs:1483:17
  13:     0x7fd24d3d4c27 - <alloc[93d8057a61472300]::boxed::Box<rustc_driver_impl[c9be22246af3610]::install_ice_hook::{closure#1}> as core[e58e1890be41adad]::ops::function::Fn<(&dyn for<'a, 'b> core[e58e1890be41adad]::ops::function::Fn<(&'a std[4f619bdb2e663cc]::panic::PanicHookInfo<'b>,), Output = ()> + core[e58e1890be41adad]::marker::Send + core[e58e1890be41adad]::marker::Sync, &std[4f619bdb2e663cc]::panic::PanicHookInfo)>>::call
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/alloc/src/boxed.rs:2254:9
  14:     0x7fd24d3d4c27 - std[4f619bdb2e663cc]::panicking::update_hook::<alloc[93d8057a61472300]::boxed::Box<rustc_driver_impl[c9be22246af3610]::install_ice_hook::{closure#1}>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:235:47
  15:     0x7fd25093283a - <alloc[93d8057a61472300]::boxed::Box<dyn for<'a, 'b> core[e58e1890be41adad]::ops::function::Fn<(&'a std[4f619bdb2e663cc]::panic::PanicHookInfo<'b>,), Output = ()> + core[e58e1890be41adad]::marker::Send + core[e58e1890be41adad]::marker::Sync> as core[e58e1890be41adad]::ops::function::Fn<(&std[4f619bdb2e663cc]::panic::PanicHookInfo,)>>::call
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/alloc/src/boxed.rs:2254:9
  16:     0x7fd25093283a - std[4f619bdb2e663cc]::panicking::panic_with_hook
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:833:13
  17:     0x7fd25090837a - std[4f619bdb2e663cc]::panicking::panic_handler::{closure#0}
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:691:13
  18:     0x7fd2508f70f9 - std[4f619bdb2e663cc]::sys::backtrace::__rust_end_short_backtrace::<std[4f619bdb2e663cc]::panicking::panic_handler::{closure#0}, !>
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/sys/backtrace.rs:182:18
  19:     0x7fd25090a29d - __rustc[bc94a6416cf16b4b]::rust_begin_unwind
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:689:5
  20:     0x7fd24bee791c - core[e58e1890be41adad]::panicking::panic_fmt
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/panicking.rs:80:14
  21:     0x7fd24bee78e2 - core[e58e1890be41adad]::panicking::panic
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/panicking.rs:150:5
  22:     0x7fd24bee75a9 - core[e58e1890be41adad]::option::unwrap_failed
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/option.rs:2251:5
  23:     0x7fd2507eba76 - rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of
  24:     0x7fd24f98f4f1 - rustc_query_impl[f274b0df531ec29f]::query_impl::layout_of::invoke_provider_fn::__rust_begin_short_backtrace
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/query_impl.rs:116:46
  25:     0x7fd24f747560 - rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr::<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:395:59
  26:     0x7fd24f747560 - rustc_middle[e976332cbd676520]::ty::context::tls::enter_context::<rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:56:9
  27:     0x7fd24f747560 - <std[4f619bdb2e663cc]::thread::local::LocalKey<core[e58e1890be41adad]::cell::Cell<*const ()>>>::try_with::<rustc_middle[e976332cbd676520]::ty::context::tls::enter_context<rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/thread/local.rs:462:12
  28:     0x7fd24f747560 - <std[4f619bdb2e663cc]::thread::local::LocalKey<core[e58e1890be41adad]::cell::Cell<*const ()>>>::with::<rustc_middle[e976332cbd676520]::ty::context::tls::enter_context<rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/thread/local.rs:426:20
  29:     0x7fd24f747560 - rustc_middle[e976332cbd676520]::ty::context::tls::enter_context::<rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:53:9
  30:     0x7fd24f747560 - rustc_query_impl[f274b0df531ec29f]::plumbing::start_query::<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/plumbing.rs:74:9
  31:     0x7fd24f747560 - rustc_middle[e976332cbd676520]::ty::context::tls::with_context::<rustc_query_impl[f274b0df531ec29f]::plumbing::start_query<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:86:36
  32:     0x7fd24f747560 - rustc_middle[e976332cbd676520]::ty::context::tls::with_context_opt::<rustc_middle[e976332cbd676520]::ty::context::tls::with_context<rustc_query_impl[f274b0df531ec29f]::plumbing::start_query<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:75:18
  33:     0x7fd24f747560 - rustc_middle[e976332cbd676520]::ty::context::tls::with_context::<rustc_query_impl[f274b0df531ec29f]::plumbing::start_query<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:86:5
  34:     0x7fd24f747560 - rustc_query_impl[f274b0df531ec29f]::plumbing::start_query::<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/plumbing.rs:61:5
  35:     0x7fd24f747560 - rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr::<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:395:17
  36:     0x7fd24f747560 - rustc_query_impl[f274b0df531ec29f]::execution::try_execute_query::<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>, false>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:303:17
  37:     0x7fd24f98f894 - rustc_query_impl[f274b0df531ec29f]::execution::execute_query_non_incr_inner::<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:596:32
  38:     0x7fd24f98f894 - stacker[ad9f6edc0433b920]::maybe_grow::<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_query_non_incr_inner<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  39:     0x7fd24f98f894 - rustc_data_structures[97dd880033bbc9ee]::stack::ensure_sufficient_stack::<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_query_non_incr_inner<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_data_structures/src/stack.rs:21:5
  40:     0x7fd24f98f894 - rustc_query_impl[f274b0df531ec29f]::execution::execute_query_non_incr_inner::<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:596:5
  41:     0x7fd24f98f894 - rustc_query_impl[f274b0df531ec29f]::query_impl::layout_of::execute_query_non_incr::__rust_end_short_backtrace
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/query_impl.rs:80:30
  42:     0x7fd250865717 - rustc_middle[e976332cbd676520]::query::inner::query_get_at::<rustc_middle[e976332cbd676520]::query::caches::DefaultCache<rustc_middle[e976332cbd676520]::ty::PseudoCanonicalInput<rustc_middle[e976332cbd676520]::ty::Ty>, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 16usize]>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/query/inner.rs:45:17
  43:     0x7fd250865717 - <rustc_middle[e976332cbd676520]::query::plumbing::TyCtxtAt>::layout_of
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/query/plumbing.rs:571:46
  44:     0x7fd250865717 - <rustc_middle[e976332cbd676520]::ty::layout::LayoutCx as rustc_middle[e976332cbd676520]::ty::layout::LayoutOf>::spanned_layout_of
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/layout.rs:704:17
  45:     0x7fd2507f59bb - <rustc_middle[e976332cbd676520]::ty::layout::LayoutCx as rustc_middle[e976332cbd676520]::ty::layout::LayoutOf>::layout_of
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/layout.rs:691:14
  46:     0x7fd2507f59bb - rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_ty_utils/src/layout.rs:653:41
  47:     0x7fd2507f59bb - core[e58e1890be41adad]::iter::adapters::map::map_try_fold::<&rustc_middle[e976332cbd676520]::ty::FieldDef, core[e58e1890be41adad]::result::Result<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>, &rustc_middle[e976332cbd676520]::ty::layout::LayoutError>, (), core[e58e1890be41adad]::ops::control_flow::ControlFlow<core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>, rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}, <core[e58e1890be41adad]::iter::adapters::GenericShunt<core[e58e1890be41adad]::iter::adapters::by_ref_sized::ByRefSized<core[e58e1890be41adad]::iter::adapters::map::Map<core[e58e1890be41adad]::slice::iter::Iter<rustc_middle[e976332cbd676520]::ty::FieldDef>, rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}>>, core[e58e1890be41adad]::result::Result<core[e58e1890be41adad]::convert::Infallible, &rustc_middle[e976332cbd676520]::ty::layout::LayoutError>> as core[e58e1890be41adad]::iter::traits::iterator::Iterator>::try_fold<(), core[e58e1890be41adad]::iter::traits::iterator::Iterator::try_for_each::call<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>::Break>::{closure#0}, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>::{closure#0}>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/iter/adapters/map.rs:95:28
  48:     0x7fd2507f59bb - <core[e58e1890be41adad]::slice::iter::Iter<rustc_middle[e976332cbd676520]::ty::FieldDef> as core[e58e1890be41adad]::iter::traits::iterator::Iterator>::try_fold::<(), core[e58e1890be41adad]::iter::adapters::map::map_try_fold<&rustc_middle[e976332cbd676520]::ty::FieldDef, core[e58e1890be41adad]::result::Result<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>, &rustc_middle[e976332cbd676520]::ty::layout::LayoutError>, (), core[e58e1890be41adad]::ops::control_flow::ControlFlow<core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>, rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}, <core[e58e1890be41adad]::iter::adapters::GenericShunt<core[e58e1890be41adad]::iter::adapters::by_ref_sized::ByRefSized<core[e58e1890be41adad]::iter::adapters::map::Map<core[e58e1890be41adad]::slice::iter::Iter<rustc_middle[e976332cbd676520]::ty::FieldDef>, rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}>>, core[e58e1890be41adad]::result::Result<core[e58e1890be41adad]::convert::Infallible, &rustc_middle[e976332cbd676520]::ty::layout::LayoutError>> as core[e58e1890be41adad]::iter::traits::iterator::Iterator>::try_fold<(), core[e58e1890be41adad]::iter::traits::iterator::Iterator::try_for_each::call<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>::Break>::{closure#0}, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>::{closure#0}>::{closure#0}, core[e58e1890be41adad]::ops::control_flow::ControlFlow<core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/iter/traits/iterator.rs:2511:21
  49:     0x7fd2507f59bb - <core[e58e1890be41adad]::iter::adapters::map::Map<core[e58e1890be41adad]::slice::iter::Iter<rustc_middle[e976332cbd676520]::ty::FieldDef>, rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}> as core[e58e1890be41adad]::iter::traits::iterator::Iterator>::try_fold::<(), <core[e58e1890be41adad]::iter::adapters::GenericShunt<core[e58e1890be41adad]::iter::adapters::by_ref_sized::ByRefSized<core[e58e1890be41adad]::iter::adapters::map::Map<core[e58e1890be41adad]::slice::iter::Iter<rustc_middle[e976332cbd676520]::ty::FieldDef>, rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}>>, core[e58e1890be41adad]::result::Result<core[e58e1890be41adad]::convert::Infallible, &rustc_middle[e976332cbd676520]::ty::layout::LayoutError>> as core[e58e1890be41adad]::iter::traits::iterator::Iterator>::try_fold<(), core[e58e1890be41adad]::iter::traits::iterator::Iterator::try_for_each::call<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>::Break>::{closure#0}, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>::{closure#0}, core[e58e1890be41adad]::ops::control_flow::ControlFlow<core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/iter/adapters/map.rs:121:19
  50:     0x7fd2507f59bb - <core[e58e1890be41adad]::iter::adapters::by_ref_sized::ByRefSized<core[e58e1890be41adad]::iter::adapters::map::Map<core[e58e1890be41adad]::slice::iter::Iter<rustc_middle[e976332cbd676520]::ty::FieldDef>, rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}>> as core[e58e1890be41adad]::iter::traits::iterator::Iterator>::try_fold::<(), <core[e58e1890be41adad]::iter::adapters::GenericShunt<core[e58e1890be41adad]::iter::adapters::by_ref_sized::ByRefSized<core[e58e1890be41adad]::iter::adapters::map::Map<core[e58e1890be41adad]::slice::iter::Iter<rustc_middle[e976332cbd676520]::ty::FieldDef>, rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}>>, core[e58e1890be41adad]::result::Result<core[e58e1890be41adad]::convert::Infallible, &rustc_middle[e976332cbd676520]::ty::layout::LayoutError>> as core[e58e1890be41adad]::iter::traits::iterator::Iterator>::try_fold<(), core[e58e1890be41adad]::iter::traits::iterator::Iterator::try_for_each::call<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>::Break>::{closure#0}, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>::{closure#0}, core[e58e1890be41adad]::ops::control_flow::ControlFlow<core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/iter/adapters/by_ref_sized.rs:54:9
  51:     0x7fd2507f59bb - <core[e58e1890be41adad]::iter::adapters::GenericShunt<core[e58e1890be41adad]::iter::adapters::by_ref_sized::ByRefSized<core[e58e1890be41adad]::iter::adapters::map::Map<core[e58e1890be41adad]::slice::iter::Iter<rustc_middle[e976332cbd676520]::ty::FieldDef>, rustc_ty_utils[a8e4f9e9a7aab2c]::layout::layout_of_uncached::{closure#23}::{closure#0}>>, core[e58e1890be41adad]::result::Result<core[e58e1890be41adad]::convert::Infallible, &rustc_middle[e976332cbd676520]::ty::layout::LayoutError>> as core[e58e1890be41adad]::iter::traits::iterator::Iterator>::try_fold::<(), core[e58e1890be41adad]::iter::traits::iterator::Iterator::try_for_each::call<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>::Break>::{closure#0}, core[e58e1890be41adad]::ops::control_flow::ControlFlow<rustc_abi[e2095b3bdd0c3997]::layout::ty::TyAndLayout<rustc_middle[e976332cbd676520]::ty::Ty>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/iter/adapters/mod.rs:195:14

< snip >

 150:     0x7fd24f989f57 - rustc_query_impl[f274b0df531ec29f]::query_impl::analysis::invoke_provider_fn::__rust_begin_short_backtrace
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/query_impl.rs:116:46
 151:     0x7fd24f731fe4 - rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr::<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:395:59
 152:     0x7fd24f731fe4 - rustc_middle[e976332cbd676520]::ty::context::tls::enter_context::<rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:56:9
 153:     0x7fd24f731fe4 - <std[4f619bdb2e663cc]::thread::local::LocalKey<core[e58e1890be41adad]::cell::Cell<*const ()>>>::try_with::<rustc_middle[e976332cbd676520]::ty::context::tls::enter_context<rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/thread/local.rs:462:12
 154:     0x7fd24f731fe4 - <std[4f619bdb2e663cc]::thread::local::LocalKey<core[e58e1890be41adad]::cell::Cell<*const ()>>>::with::<rustc_middle[e976332cbd676520]::ty::context::tls::enter_context<rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/thread/local.rs:426:20
 155:     0x7fd24f731fe4 - rustc_middle[e976332cbd676520]::ty::context::tls::enter_context::<rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:53:9
 156:     0x7fd24f731fe4 - rustc_query_impl[f274b0df531ec29f]::plumbing::start_query::<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/plumbing.rs:74:9
 157:     0x7fd24f731fe4 - rustc_middle[e976332cbd676520]::ty::context::tls::with_context::<rustc_query_impl[f274b0df531ec29f]::plumbing::start_query<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:86:36
 158:     0x7fd24f731fe4 - rustc_middle[e976332cbd676520]::ty::context::tls::with_context_opt::<rustc_middle[e976332cbd676520]::ty::context::tls::with_context<rustc_query_impl[f274b0df531ec29f]::plumbing::start_query<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:75:18
 159:     0x7fd24f731fe4 - rustc_middle[e976332cbd676520]::ty::context::tls::with_context::<rustc_query_impl[f274b0df531ec29f]::plumbing::start_query<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}>::{closure#0}, rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:86:5
 160:     0x7fd24f731fe4 - rustc_query_impl[f274b0df531ec29f]::plumbing::start_query::<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/plumbing.rs:61:5
 161:     0x7fd24f731fe4 - rustc_query_impl[f274b0df531ec29f]::execution::execute_job_non_incr::<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:395:17
 162:     0x7fd24f731fe4 - rustc_query_impl[f274b0df531ec29f]::execution::try_execute_query::<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>, false>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:303:17
 163:     0x7fd24f98a2f1 - rustc_query_impl[f274b0df531ec29f]::execution::execute_query_non_incr_inner::<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:596:32
 164:     0x7fd24f98a2f1 - stacker[ad9f6edc0433b920]::maybe_grow::<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_query_non_incr_inner<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
 165:     0x7fd24f98a2f1 - rustc_data_structures[97dd880033bbc9ee]::stack::ensure_sufficient_stack::<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>, rustc_query_impl[f274b0df531ec29f]::execution::execute_query_non_incr_inner<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>::{closure#0}>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_data_structures/src/stack.rs:21:5
 166:     0x7fd24f98a2f1 - rustc_query_impl[f274b0df531ec29f]::execution::execute_query_non_incr_inner::<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/execution.rs:596:5
 167:     0x7fd24f98a2f1 - rustc_query_impl[f274b0df531ec29f]::query_impl::analysis::execute_query_non_incr::__rust_end_short_backtrace
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_query_impl/src/query_impl.rs:80:30
 168:     0x7fd24d35d0ef - rustc_middle[e976332cbd676520]::query::inner::query_ensure_ok_or_done::<rustc_middle[e976332cbd676520]::query::caches::SingleCache<rustc_middle[e976332cbd676520]::query::erase::ErasedData<[u8; 0usize]>>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/query/inner.rs:63:13
 169:     0x7fd24d35d0ef - <rustc_middle[e976332cbd676520]::query::plumbing::TyCtxtEnsureOk>::analysis
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/query/plumbing.rs:586:21
 170:     0x7fd24d35d0ef - rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_driver_impl/src/lib.rs:325:29
 171:     0x7fd24d35d0ef - rustc_interface[8ee6c83dc6f50ddf]::passes::create_and_enter_global_ctxt::<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}>::{closure#2}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_interface/src/passes.rs:1019:23
 172:     0x7fd24d35d0ef - <rustc_middle[e976332cbd676520]::ty::context::GlobalCtxt>::enter::<rustc_interface[8ee6c83dc6f50ddf]::passes::create_and_enter_global_ctxt<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}>::{closure#2}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>::{closure#1}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context.rs:863:37
 173:     0x7fd24d35d0ef - rustc_middle[e976332cbd676520]::ty::context::tls::enter_context::<<rustc_middle[e976332cbd676520]::ty::context::GlobalCtxt>::enter<rustc_interface[8ee6c83dc6f50ddf]::passes::create_and_enter_global_ctxt<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}>::{closure#2}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>::{closure#1}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:56:9
 174:     0x7fd24d35d0ef - <std[4f619bdb2e663cc]::thread::local::LocalKey<core[e58e1890be41adad]::cell::Cell<*const ()>>>::try_with::<rustc_middle[e976332cbd676520]::ty::context::tls::enter_context<<rustc_middle[e976332cbd676520]::ty::context::GlobalCtxt>::enter<rustc_interface[8ee6c83dc6f50ddf]::passes::create_and_enter_global_ctxt<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}>::{closure#2}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>::{closure#1}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>::{closure#0}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/thread/local.rs:462:12
 175:     0x7fd24d35d0ef - <std[4f619bdb2e663cc]::thread::local::LocalKey<core[e58e1890be41adad]::cell::Cell<*const ()>>>::with::<rustc_middle[e976332cbd676520]::ty::context::tls::enter_context<<rustc_middle[e976332cbd676520]::ty::context::GlobalCtxt>::enter<rustc_interface[8ee6c83dc6f50ddf]::passes::create_and_enter_global_ctxt<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}>::{closure#2}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>::{closure#1}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>::{closure#0}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/thread/local.rs:426:20
 176:     0x7fd24d35d0ef - rustc_middle[e976332cbd676520]::ty::context::tls::enter_context::<<rustc_middle[e976332cbd676520]::ty::context::GlobalCtxt>::enter<rustc_interface[8ee6c83dc6f50ddf]::passes::create_and_enter_global_ctxt<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}>::{closure#2}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>::{closure#1}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context/tls.rs:53:9
 177:     0x7fd24d35d0ef - <rustc_middle[e976332cbd676520]::ty::context::GlobalCtxt>::enter::<rustc_interface[8ee6c83dc6f50ddf]::passes::create_and_enter_global_ctxt<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}>::{closure#2}, core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context.rs:863:9
 178:     0x7fd24d35d0ef - <rustc_middle[e976332cbd676520]::ty::context::TyCtxt>::create_global_ctxt::<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_interface[8ee6c83dc6f50ddf]::passes::create_and_enter_global_ctxt<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}>::{closure#2}>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_middle/src/ty/context.rs:1072:13
 179:     0x7fd24d35d0ef - rustc_interface[8ee6c83dc6f50ddf]::passes::create_and_enter_global_ctxt::<core[e58e1890be41adad]::option::Option<rustc_interface[8ee6c83dc6f50ddf]::queries::Linker>, rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}::{closure#2}>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_interface/src/passes.rs:986:5
 180:     0x7fd24d3d39a5 - rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_driver_impl/src/lib.rs:298:22
 181:     0x7fd24d3d39a5 - rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler::<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_interface/src/interface.rs:500:80
 182:     0x7fd24d3d39a5 - <core[e58e1890be41adad]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}::{closure#0}> as core[e58e1890be41adad]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/panic/unwind_safe.rs:275:9
 183:     0x7fd24d3d39a5 - std[4f619bdb2e663cc]::panicking::catch_unwind::do_call::<core[e58e1890be41adad]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:581:40
 184:     0x7fd24d3d39a5 - std[4f619bdb2e663cc]::panicking::catch_unwind::<(), core[e58e1890be41adad]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}::{closure#0}>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:544:19
 185:     0x7fd24d3d39a5 - std[4f619bdb2e663cc]::panic::catch_unwind::<core[e58e1890be41adad]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panic.rs:359:14
 186:     0x7fd24d3d39a5 - rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler::<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_interface/src/interface.rs:500:23
 187:     0x7fd24d3c3125 - rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals::<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_interface/src/util.rs:203:17
 188:     0x7fd24d3c3125 - rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals::<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_interface/src/util.rs:159:24
 189:     0x7fd24d3c3125 - <scoped_tls[aa3cf235b5f1b553]::ScopedKey<rustc_span[7d7c86df1226e373]::SessionGlobals>>::set::<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}, ()>
                               at /rust/deps/scoped-tls-1.0.1/src/lib.rs:137:9
 190:     0x7fd24d3c3125 - rustc_span[7d7c86df1226e373]::create_session_globals_then::<(), rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_span/src/lib.rs:153:21
 191:     0x7fd24d3c3125 - rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals::<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/compiler/rustc_interface/src/util.rs:155:17
 192:     0x7fd24d3c3125 - std[4f619bdb2e663cc]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/sys/backtrace.rs:166:18
 193:     0x7fd24d3da5d3 - std[4f619bdb2e663cc]::thread::lifecycle::spawn_unchecked::<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/thread/lifecycle.rs:91:13
 194:     0x7fd24d3da5d3 - <core[e58e1890be41adad]::panic::unwind_safe::AssertUnwindSafe<std[4f619bdb2e663cc]::thread::lifecycle::spawn_unchecked<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}> as core[e58e1890be41adad]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/panic/unwind_safe.rs:275:9
 195:     0x7fd24d3da5d3 - std[4f619bdb2e663cc]::panicking::catch_unwind::do_call::<core[e58e1890be41adad]::panic::unwind_safe::AssertUnwindSafe<std[4f619bdb2e663cc]::thread::lifecycle::spawn_unchecked<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:581:40
 196:     0x7fd24d3da5d3 - std[4f619bdb2e663cc]::panicking::catch_unwind::<(), core[e58e1890be41adad]::panic::unwind_safe::AssertUnwindSafe<std[4f619bdb2e663cc]::thread::lifecycle::spawn_unchecked<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panicking.rs:544:19
 197:     0x7fd24d3da5d3 - std[4f619bdb2e663cc]::panic::catch_unwind::<core[e58e1890be41adad]::panic::unwind_safe::AssertUnwindSafe<std[4f619bdb2e663cc]::thread::lifecycle::spawn_unchecked<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/panic.rs:359:14
 198:     0x7fd24d3da5d3 - std[4f619bdb2e663cc]::thread::lifecycle::spawn_unchecked::<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/thread/lifecycle.rs:89:26
 199:     0x7fd24d3da5d3 - <std[4f619bdb2e663cc]::thread::lifecycle::spawn_unchecked<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_with_globals<rustc_interface[8ee6c83dc6f50ddf]::util::run_in_thread_pool_with_globals<rustc_interface[8ee6c83dc6f50ddf]::interface::run_compiler<(), rustc_driver_impl[c9be22246af3610]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[e58e1890be41adad]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
                               at /rustc-dev/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/core/src/ops/function.rs:250:5
 200:     0x7fd25094063f - <alloc[93d8057a61472300]::boxed::Box<dyn core[e58e1890be41adad]::ops::function::FnOnce<(), Output = ()> + core[e58e1890be41adad]::marker::Send> as core[e58e1890be41adad]::ops::function::FnOnce<()>>::call_once
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/alloc/src/boxed.rs:2240:9
 201:     0x7fd25094063f - <std[4f619bdb2e663cc]::sys::thread::unix::Thread>::new::thread_start
                               at /rustc/3a933e5bf5b695faf9e09d47a263a20c6b85c26a/library/std/src/sys/thread/unix.rs:118:17
 202:     0x7fd24a6a597a - <unknown>
 203:     0x7fd24a7292bc - <unknown>
 204:                0x0 - <unknown>

error: the compiler unexpectedly panicked. This is a bug

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.96.0-nightly (3a933e5bf 2026-03-26) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [layout_of] computing layout of `unsafe<> core::option::Option<alloc::boxed::Box<dyn core::marker::Send>>`
#1 [layout_of] computing layout of `core::option::Option<unsafe<> core::option::Option<alloc::boxed::Box<dyn core::marker::Send>>>`
#2 [mir_drops_elaborated_and_const_checked] elaborating drops for `main`
#3 [analysis] running analysis passes on crate `mvce`
end of query stack
error: aborting due to 3 previous errors

Some errors have detailed explanations: E0277, E0658.
For more information about an error, try `rustc --explain E0277`.

```

</p>
</details>

<!--
query stack:
#0 [layout_of] computing layout of `unsafe<> core::option::Option<alloc::boxed::Box<dyn core::marker::Send>>`
#1 [layout_of] computing layout of `core::option::Option<unsafe<> core::option::Option<alloc::boxed::Box<dyn core::marker::Send>>>`
#2 [mir_drops_elaborated_and_const_checked] elaborating drops for `main`
#3 [analysis] running analysis passes on crate `mvce`

-->
@rustbot label +F-unsafe_binders

--- a/compiler/rustc_middle/src/ty/sty.rs
+++ b/compiler/rustc_middle/src/ty/sty.rs
@@ -1630,6 +1630,9 @@ impl<'tcx> Ty<'tcx> {
             TyKind::Coroutine(def_id, args) => {
                 Some(args.as_coroutine().variant_range(*def_id, tcx))
             }
+            TyKind::UnsafeBinder(bound_ty) => {
+                tcx.instantiate_bound_regions_with_erased((*bound_ty).into()).variant_range(tcx)
+            }
             _ => None,
         }
     }
@@ -1651,6 +1654,9 @@ impl<'tcx> Ty<'tcx> {
             TyKind::Coroutine(def_id, args) => {
                 Some(args.as_coroutine().discriminant_for_variant(*def_id, tcx, variant_index))
             }
+            TyKind::UnsafeBinder(bound_ty) => tcx
+                .instantiate_bound_regions_with_erased((*bound_ty).into())
+                .discriminant_for_variant(tcx, variant_index),
             _ => None,
         }
     }
@@ -1669,6 +1675,9 @@ impl<'tcx> Ty<'tcx> {
             }
 
             ty::Pat(ty, _) => ty.discriminant_ty(tcx),
+            ty::UnsafeBinder(bound_ty) => {
+                tcx.instantiate_bound_regions_with_erased((*bound_ty).into()).discriminant_ty(tcx)
+            }
 
             ty::Bool
             | ty::Char
@@ -1690,7 +1699,6 @@ impl<'tcx> Ty<'tcx> {
             | ty::CoroutineWitness(..)
             | ty::Never
             | ty::Tuple(_)
-            | ty::UnsafeBinder(_)
             | ty::Error(_)
             | ty::Infer(IntVar(_) | FloatVar(_)) => tcx.types.u8,
 

--- a/tests/ui/unsafe-binders/discriminant-for-variant.rs
+++ b/tests/ui/unsafe-binders/discriminant-for-variant.rs
@@ -1,0 +1,11 @@
+#![feature(unsafe_binders)]
+
+const None: Option<unsafe<> Option<Box<dyn Send>>> = None;
+//~^ ERROR the trait bound `Box<(dyn Send + 'static)>: Copy` is not satisfied
+//~| ERROR the trait bound `Box<(dyn Send + 'static)>: Copy` is not satisfied
+
+fn main() {
+    match None {
+        _ => {}
+    }
+}

--- a/tests/ui/unsafe-binders/discriminant-for-variant.stderr
+++ b/tests/ui/unsafe-binders/discriminant-for-variant.stderr
@@ -1,0 +1,19 @@
+error[E0277]: the trait bound `Box<(dyn Send + 'static)>: Copy` is not satisfied
+  --> $DIR/discriminant-for-variant.rs:3:13
+   |
+LL | const None: Option<unsafe<> Option<Box<dyn Send>>> = None;
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `Box<(dyn Send + 'static)>`
+   |
+   = note: required for `Option<Box<(dyn Send + 'static)>>` to implement `Copy`
+
+error[E0277]: the trait bound `Box<(dyn Send + 'static)>: Copy` is not satisfied
+  --> $DIR/discriminant-for-variant.rs:3:54
+   |
+LL | const None: Option<unsafe<> Option<Box<dyn Send>>> = None;
+   |                                                      ^^^^ the trait `Copy` is not implemented for `Box<(dyn Send + 'static)>`
+   |
+   = note: required for `Option<Box<(dyn Send + 'static)>>` to implement `Copy`
+
+error: aborting due to 2 previous errors
+
+For more information about this error, try `rustc --explain E0277`.

